### PR TITLE
Improve unit testing

### PR DESCRIPTION
**Pull request by ant1g**
This PR provides the ground work for testing of the packet forwarder feature.
It contains:
- Mocking with gomock
- An updated make command to create the mocks
- A way to easily setup test dependencies (keepers, IBC modules)
- Fixed error messages
- Various refactoring & naming improvements

### CODE REVIEW

File: router/keeper/keeper.go [hunk 1]
```diff
@@ -27,11 +28,19 @@ type Keeper struct {
 	distrKeeper    types.DistributionKeeper
 }
 
+func TransferDefaultTimeout(ctx sdk.Context) uint64 {
```
Comment (jtieri) [inline router/keeper/keeper.go:31]: I know we were setting the timeout to 30 minutes before but what if we used the same defaults from ibc-go just for consistency?

https://github.com/cosmos/ibc-go/blob/06c2d764df5945d8de93870184ba38b7a8a5dc05/modules/apps/transfer/types/packet.go#L11-L21

**Review by jtieri — APPROVED** [review]
Thanks for the contribution @ant1g this is great! 

Just had the one suggestion but i'll leave you to decide if you want to change it or not.

File: router/keeper/keeper.go (hunk 1 above)
Comment (ant1g) [reply router/keeper/keeper.go]: I also used the height timeout from IBC then. Please check!

Comment (jtieri) [reply router/keeper/keeper.go]: Looks good to me! Thanks for that 🙂

**Pull request merged**